internal/checkers/http_const.go

```diff
@@ -0,0 +1,122 @@
+package checkers
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// HTTPConst detects situations like
+//
+//	assert.HTTPStatusCode(t, handler, "GET", "/index", nil, 200)
+//	assert.HTTPBodyContains(t, handler, "GET", "/index", nil, "counter")
+//
+// and requires
+//
+//	assert.HTTPStatusCode(t, handler, http.MethodGet, "/index", nil, http.StatusOK)
+//	assert.HTTPBodyContains(t, handler, http.MethodGet, "/index", nil, "counter")
+type HTTPConst struct{}
+
+// NewHTTPConst constructs Regexp checker.
+func NewHTTPConst() HTTPConst  { return HTTPConst{} }
+func (HTTPConst) Name() string { return "http-const" }
+
+func (checker HTTPConst) Check(_ *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	switch call.Fn.NameFTrimmed {
+	case "HTTPBody",
+		"HTTPBodyContains",
+		"HTTPBodyContainsf",
+		"HTTPBodyNotContains",
+		"HTTPBodyNotContainsf",
+		"HTTPError",
+		"HTTPErrorf",
+		"HTTPRedirect",
```

Comment:
No HTTPRedirectf ?

https://pkg.go.dev/github.com/stretchr/testify/assert#HTTPRedirectf

Reply:
Wait, you are using trimmed variable, please remove all the one with a trailing f